specification/vmware/Microsoft.AVS/routes.tsp

```diff
@@ -0,0 +1,970 @@
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/publicIPs/{publicIPId}")
+  @doc("Delete a {name}", WorkloadNetworkPublicIP)
+  @delete
+  @extension("x-ms-long-running-operation", true)
+  @extension(
+    "x-ms-long-running-operation-options",
+    {
+      `final-state-via`: "location",
+    }
+  )
+  delete(
+    ...BaseParameters<WorkloadNetworkPublicIP>,
+    ...PublicIPIdParameter,
+    ...PrivateCloudNameParameter,
+  ):
+    | ArmDeletedResponse
+    | ArmDeleteAcceptedLroResponse
+    | ArmDeletedNoContentResponse
+    | ErrorResponse;
+}
+
+@@projectedName(WorkloadNetworkPublicIps.create::parameters.resource,
+  "json",
+  "workloadNetworkPublicIP"
+);
+@@OpenAPI.extension(WorkloadNetworkPublicIps.create::parameters.resource,
+  "x-ms-client-name",
+  "workloadNetworkPublicIP"
+);
+
+@armResourceOperations
+interface CloudLinks {
+  #suppress "@azure-tools/typespec-azure-core/no-operation-id" "Can not change existing operationId."
+  @operationId("CloudLinks_List")
+  list is ArmResourceListByParent<CloudLink>;
+
+  get is ArmResourceRead<CloudLink>;
+
+  createOrUpdate is ArmResourceCreateOrUpdateAsync<CloudLink>;
+
+  #suppress "deprecated"
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes"
+  delete is ArmResourceDeleteAsync<CloudLink>;
+}
+
+@@projectedName(CloudLinks.createOrUpdate::parameters.resource,
+  "json",
+  "cloudLink"
+);
+@@OpenAPI.extension(CloudLinks.createOrUpdate::parameters.resource,
+  "x-ms-client-name",
+  "cloudLink"
+);
+
+@armResourceOperations
+interface Addons {
+  #suppress "@azure-tools/typespec-azure-core/no-operation-id" "Can not change existing operationId."
+  @operationId("Addons_List")
+  list is ArmResourceListByParent<Addon>;
+
+  get is ArmResourceRead<Addon>;
+
+  createOrUpdate is ArmResourceCreateOrUpdateAsync<Addon>;
+
+  #suppress "deprecated"
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes"
+  delete is ArmResourceDeleteAsync<Addon>;
+}
+
+@@projectedName(Addons.createOrUpdate::parameters.resource, "json", "addon");
+@@OpenAPI.extension(Addons.createOrUpdate::parameters.resource,
+  "x-ms-client-name",
+  "addon"
+);
+
+@armResourceOperations
+interface VirtualMachines {
+  #suppress "@azure-tools/typespec-azure-core/no-operation-id" "Can not change existing operationId."
+  @operationId("VirtualMachines_List")
+  list is ArmResourceListByParent<VirtualMachine>;
+
+  get is ArmResourceRead<VirtualMachine>;
+
+  // restrictMovement should be a ArmResourceActionNoContentAsync, but does not respond with a 204.
+  @doc("Enable or disable DRS-driven VM movement restriction")
+  @armResourceAction(Location)
+  @post
+  @extension("x-ms-long-running-operation", true)
+  @extension(
+    "x-ms-long-running-operation-options",
+    {
+      `final-state-via`: "azure-async-operation",
```

Comment:
May be `final-state-via` should be `location`.

> location - poll until terminal state, if the initial response had a Location header, a final GET will be done. Default behavior for POST operation.
https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-long-running-operation-options

Reply:
I looks like the default behavior for POST is location. Out of caution, I'm simply removing unsetting it, same as previous API version.